### PR TITLE
Publish NuGet packages for 15.6

### PR DESCRIPTION
**Pull request by jaredpar**
Updating our publishing story for the 15.6 release:

- Moving to the 2.7 version number scheme
- Again publish packages out of master

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-infrastructure

File: build/config/PublishData.json [hunk 1]
```diff
@@ -1,18 +1,18 @@
 {
     "branches": {
-        "dev15.5.x": {
+        "master": {
```
Comment (jasonmalinowski) [inline build/config/PublishData.json]: This should be master-vs-deps, as that's where we'll be publishing packages.

File: build/Targets/Versions.props [hunk 2]
```diff
@@ -8,13 +8,13 @@
...
-    <RoslynAssemblyVersionBase Condition="'$(RoslynAssemblyVersion)' == ''">2.6.0</RoslynAssemblyVersionBase>
+    <RoslynAssemblyVersionBase Condition="'$(RoslynAssemblyVersion)' == ''">2.7.0</RoslynAssemblyVersionBase>
     <!-- This is the file version of Roslyn, as placed in the PE header. It should be revved during point releases, and is also what provides the basis for our NuGet package versioning. -->
-    <RoslynFileVersionBase Condition="'$(RoslynFileVersionBase)' == ''">2.6.0</RoslynFileVersionBase>
+    <RoslynFileVersionBase Condition="'$(RoslynFileVersionBase)' == ''">2.7.0</RoslynFileVersionBase>
     <!-- The release moniker for our packages.  Developers should use "dev" and official builds pick the branch
          moniker listed below -->
     <RoslynNuGetMoniker Condition="'$(RoslynNuGetMoniker)' == ''">dev</RoslynNuGetMoniker>
-    <RoslynNuGetMoniker Condition="'$(OfficialBuild)' == 'true'">beta3</RoslynNuGetMoniker> 
+    <RoslynNuGetMoniker Condition="'$(OfficialBuild)' == 'true'">beta1</RoslynNuGetMoniker>
```
Comment (jasonmalinowski) [inline build/Targets/Versions.props]: beta2 now that this branch is Preview 2?

File: build/config/PublishData.json (hunk 1 above)
Comment (jasonmalinowski) [reply build/config/PublishData.json]: (or alternately: if you *do* want to publish out of master, then the channels needs some way to distinguish)

**Comment by jaredpar** [comment]
Responded to feedback

**Comment by shyamnamboodiripad** [comment]
cc @genlu

**Comment by jaredpar** [comment]
@jasonmalinowski okay now that feedback addressed?

File: build/config/PublishData.json [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
     "branches": {
-        "dev15.5.x": {
+        "master-vs-deps": {
             "nugetKind": "PerBuildPreRelease",
-            "version": "2.6.*",
+            "version": "2.7.*",
             "nuget": [ "https://dotnet.myget.org/F/roslyn/api/v2/package" ],
             "vsix": [ "https://dotnet.myget.org/F/roslyn/vsix/upload" ],
-            "channels": [ "dev15.5", "dev15.5p3" ]
+            "channels": [ "dev15.6" ]
```
Comment (genlu) [inline build/config/PublishData.json]: Just curious, what does the `channel` do?

Comment (jaredpar) [reply build/config/PublishData.json]: This updates value in the dotnet/versions repo.

**Pull request merged**